Projects/Core/Services/UserNotification/UserNotificationClient.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  UserNotificationClient.swift
+//  Service
+//
+//  Created by YangJoonHyeok on 6/8/24.
+//  Copyright © 2024 com.mashup.gabbangzip. All rights reserved.
+//
+
+import ComposableArchitecture
+import UserNotifications
+
+public struct UserNotificationClient {
+  public var getAuthorizationStatus: @Sendable () async -> UNAuthorizationStatus
+  public var delegate: @Sendable () -> AsyncStream<DelegateEvent>
+  public var requestAuthorization: @Sendable () -> Void
+  
+  public enum DelegateEvent {
+    case didReceiveResponse(
+      UNNotificationResponse,
+      completionHandler: @Sendable () -> Void
+    )
+    case willPresentNotification(
+      UNNotification,
+      completionHandler: @Sendable (
+        UNNotificationPresentationOptions
+      ) -> Void
+    )
+  }
+}
+
+extension UserNotificationClient: DependencyKey {
+  public static var liveValue: UserNotificationClient {
+    let notificationCenter = UNUserNotificationCenter.current()
+    return .init(
+      getAuthorizationStatus: {
+        let authorizationStatus = await notificationCenter.notificationSettings().authorizationStatus
+        return authorizationStatus
+      },
+      delegate: {
+        AsyncStream { continuation in
+          let delegate = UserNotificationClient.Delegate(continuation: continuation)
```

Comment:
```suggestion
          let delegate = Delegate(continuation: continuation)
```
p2. Delegate 클래스가 구조체 내부에 정의되어 중첩되 있기에 해당 구조체를 명시적으로 붙여서 사용하지 않아도 충분할것 같습니다 🙋🏻
해당 Delegate를 사용하는 외부 타입에서 UserNotificationClient의 Delegate를 접근해야한다면 필요할 순 있지만 내부에서는 Delegate로 사용하는것이 외부에서 직접적인 생성 및 접근을 막을 수 있어 캡슐화를 강화할 수 있지 않을까해요.

Reply:
그렇군요😯 말씀하신 대로 앞에 UserNotificationClient를 빼고 Delegate로 시작하는 것이 더 좋아보입니다